File: Algorithms/Searching/linear_search.cpp
#include <iostream>
using namespace std;

int linear_search(int arr[], int size, int target)
{
    for (int i = 0; i < size; i++)
    {
        if (arr[i] == target)
        {
            return i; // FOUND
        }
    }
    return -1; // NOT FOUND
}

int main()
{
    int arr[] = {4, 2, 7, 8, 1, 2, 5};
    int size = sizeof(arr) / sizeof(int);

    int target;
    cout << "Enter you Target = ";
    cin >> target;

    int index = linear_search(arr, size, target);

    if (index == -1)
    {
        cout << "Element not found" << endl;
    }

    else
    {
        cout << "Element Found at Position = " << index << endl;
    }

    return 0;
}

File: python/Algorithms/Searching/linear_search.py
def linear_search(arr, target):
    for index, number in enumerate(arr):
        if number == target:
            return index
    
    return -1
    
    
arr = [4, 2, 7, 8, 1, 2, 5]

target = int(input("Enter the Target :- "))
index = linear_search(arr, target)

if index != -1:
    print("Index Postiton :- ",index)
else:
    print("Target not found, the array is :- ", arr)